### PR TITLE
Clean branch for API Streamer docs

### PR DESCRIPTION
**Pull request by Srushtika**

### CODE REVIEW

File: content/api-streamer/overview.textile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+h2(#what-is-api-streamer). What is API Streamer?
+
+"API Streamer":https://www.ably.io/api-streamer makes it easy for developers to publish realtime APIs (also known as event-driven APIs or streaming APIs). Built on <a href="https://www.ably.io/dsn">Ably's global Data Stream Network</a>, API Streamer provides an additional layer of tooling and features for deploying streaming data to the Ably network, managing data streams, and distributing those data streams to developer teams at third party organizations through realtime push and pull subscription APIs.
+
+<a href="https://www.ably.io/api-streamer">
+  <img src="/images/diagrams/api-streamer-overview.png" style="width: 100%" alt="Ably API Streamer">
+</a>
+
+h3(#why-use-api-streamer). Why use the API Streamer to build realtime APIs for your streaming data?
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: I think it's less a `why`, more of a `what` are the use cases?

File: content/api-streamer/overview.textile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Many early adopter organizations are addressing this problem by building their own realtime APIs, typically based on WebSockets, WebHooks or a custom streaming HTTP protocol. These organizations are realizing that delivering realtime data at scale, with low latency, across the current and emerging protocols, with the necessary tooling to manage these data flows, is a significant upfront and ongoing engineering undertaking.
+
+This is the problem that the API Streamer addresses. Developers can focus on the data that needs to be shared, whilst Ably provides ready-built realtime API management tools, a global low latency and reliable distribution network, protocol interoperability and limitless scale.
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: A few links to proof of claims here would be nice.

File: content/api-streamer/overview.textile [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Here's a look at the increasing engineering effort required with time and growing customers, when building a realtime API from scratch vs when using the API Streamer:
+
+<a href="/images/diagrams/api-streamer-effort-time-graph@3x.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-effort-time-graph@3x.png" style="width: 100%" alt="API Streamer vs building a realtime API">
+</a>
+
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: Extra line.

File: content/api-streamer/overview.textile [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Many early adopter organizations are addressing this problem by building their own realtime APIs, typically based on WebSockets, WebHooks or a custom streaming HTTP protocol. These organizations are realizing that delivering realtime data at scale, with low latency, across the current and emerging protocols, with the necessary tooling to manage these data flows, is a significant upfront and ongoing engineering undertaking.
+
+This is the problem that the API Streamer addresses. Developers can focus on the data that needs to be shared, whilst Ably provides ready-built realtime API management tools, a global low latency and reliable distribution network, protocol interoperability and limitless scale.
+
+Here's a look at the increasing engineering effort required with time and growing customers, when building a realtime API from scratch vs when using the API Streamer:
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: This is again quite sales-y for a docs article, perhaps worth thinking of how much we want it to be.

File: content/api-streamer/overview.textile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+<a href="/images/diagrams/api-streamer-effort-time-graph@3x.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-effort-time-graph@3x.png" style="width: 100%" alt="API Streamer vs building a realtime API">
+</a>
+
+
+h2(#api-streamer-roles). Understanding the working of the API Streamer
+
+Ably's established "global Data Stream Network (DSN)":https://www.ably.io/dsn offers a simple API to developers who publish and receive realtime data using "pub/sub channels":https://www.ably.io/channels. Ably is a serverless messaging platform that offers unparalleled reliability, latency, security and streaming at any scale, globally.
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: `"global Data` I'd say don't link to global part.

File: content/api-streamer/overview.textile [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+Due to the underlying pub/sub pattern in action, the API Streamer allows data producers to be decoupled from data subscribers, offloading the engineering complexity involved with scale, interoperability and distribution via the Ably network. Whilst Ably ensures producers and consumers are technically decoupled, producers retain complete control over their data with API Streamer dashboards, instrumentation, reporting and monetization features.
+
+h3(#producer). The Data Producer
+
+Data producers are typically organizations or developers who use the API Streamer to distribute their realtime data streams with other developers. Data producers publish their data over channels to the Ably DSN, and using access control lists that are configurable within the API Streamer dashboards they can manage who has access to their data. For example, a train company may publish live train information on a channel in one their Ably apps, and using the API streamer, allow developers to subscribe to this data via the Ably platform. Developers may, for example, elect to have this data pushed to them via WebSub, or subscribe to this data directly on their mobile devices.
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: Links here for channels, DSN, etc would be good.

File: content/api-streamer/overview.textile [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+"Learn more about Producers and distributing data using the API Streamer":/api-streamer/producer.
+
+h3(#consumer). The Data Consumer
+
+Data consumers are typically organizations or developers, who have been given access to one or more data streams, from one or more data producers. The consumer, using their API Streamer dashboard, can configure how they want to consume these data streams.
+
+Ably supports two models for subscribing to streams of data:
+
+* **Pull Subscription Protocols** are used by clients that connect to Ably and explicitly subscribe to streams on-demand. Protocols such as MQTT, SSE, or Ably's native protocol, are all Pull based protocols as clients initiate the connection to Ably and request a subscription to a stream of data from the Ably service
```
Comment (tomczoink) [inline content/api-streamer/overview.textile]: Missing fullstop at end.

Comment (tomczoink) [reply content/api-streamer/overview.textile]: Same pull/push queries as above in consumer page.

File: content/api-streamer/producer.textile [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+You can share your realtime data streams with consumers in three distinct steps: 1) Deploy,  2) Manage and 3) Distribute.
+
+<a href="/images/diagrams/api-streamer-producer-steps.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-producer-steps.png" style="width: 100%" alt="Steps involved in publishing data streams">
+</a>
+
+h3(#deploy). Deploy
+
+Event-driven and realtime data, by it's very nature, needs to flow from the producer to the consumer with the lowest possible latency. Given producers in most cases are decoupled from consumers of data, and not all data that is published is needed by all subscribers, the "publish/subscribe pattern":https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern is most commonly used in systems designed to deliver realtime data as streams of events or messages.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: `by it's very nature` => `by its very nature`

File: content/api-streamer/producer.textile [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+In the first step of publishing data streams, you'll need to set up an app on the Ably app dashboard or use one of the existing apps. By default, all the Ably accounts come with two pre-loaded apps called 'Production' and 'Sandbox'. If you are just trying out the API Streamer for the first time, we recommend you use the 'Sandbox' app to get up and running quickly.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: As above in consumer page, I think this needs removing/simplifying/describing elsewhere instead.

File: content/api-streamer/producer.textile [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+Ably fundamentally organizes all the data into units called "channels":/core-features/channels. In order to deploy your data into your Ably apps, you'll need to publish it to one or more channels.
+
+You can do this by either by using our "Realtime Client SDKs":/realtime, our "REST Client SDKs":/rest, our "MQTT adapter":/mqtt, or directly using our "raw REST API":/rest-api (although this last option isn't recommended as it is inferior to our SDKs in terms of performance and resilience).
+
+In most cases, if you are only publishing your data streams and not expecting a two-way communication between you and the consumer of the data, the most efficient way is to use the REST Client SDKs.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: `not expecting a two-way communication between you and the consumer of the data` don't think using streamer it's possible to have two-way communication.

File: content/api-streamer/producer.textile [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+
+You authenticate yourself using the API Key in your app, and start publishing your data into one or more channels. Additionally, if your publisher is not secure enough to embed API Keys directly, you can "set up a separate authentication server and issue tokens to your publisher":/core-features/authentication#token-authentication who can then use them to authenticate with Ably before being able to publish.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: Above consumer page discussion on this applies.

File: content/api-streamer/producer.textile [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+```[go]
+  rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+  channel := rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}", nil)
+  interval := setInterval(func() {
+            err = channel.Publish("example", "message data")
+  }, 3000, false)
+  
+```
+
+h4(#deploy-using-realtime-sdks). Deploying data streams using Ably's Realtime Client SDKs
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: Not sure if having REST + Realtime example here is necessary, given both look exactly the same besides for the most part.

File: content/api-streamer/producer.textile [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+@objc func publishData(){
+ channel.publish("example", data: "message data")
+}
+```
+
+h3(#manage). Manage
+
+After you've successfully set up your app and know how to deploy your data streams into them, you can set up products to manage those data streams. Products allow you to distribute different subsets of data streams within your apps to different consumers.
+
+In your API Streamer producer dashboard's products tab, you can create new Products. While doing so, you can select the app into which you have deployed the data streams you intend to share using that Product. You can then specify a channel namespace, which is essentially a name given to the logical subset of the data in your app that you wish you share. For example, if you are publishing transport data to one of your apps called 'Public Transport Data', you could be publishing the data related to buses into the namespace called 'bus' and the data related to trains into another namespace called 'train'. You can then create two separate products, one for each of these namespaces to be distributed to the same or different sets of consumers.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: Definitely need some links here to each concept, how to create things, selecting apps, etc.

File: content/api-streamer/producer.textile [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+
+In your API Streamer producer dashboard's products tab, you can create new Products. While doing so, you can select the app into which you have deployed the data streams you intend to share using that Product. You can then specify a channel namespace, which is essentially a name given to the logical subset of the data in your app that you wish you share. For example, if you are publishing transport data to one of your apps called 'Public Transport Data', you could be publishing the data related to buses into the namespace called 'bus' and the data related to trains into another namespace called 'train'. You can then create two separate products, one for each of these namespaces to be distributed to the same or different sets of consumers. 
+
+The concept of channels explained earlier brings further categorization to each of these namespaces. For instance, the bus data for London and Manchester could go in two channels 'bus:london' and 'bus:manchester'. A product created using the namespace 'bus' will give it's consumers access to all the channels that are part of that namespace, in this case 'london' and 'manchester'.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: For channel names, it'd be good to highlight them rather than just put in quotes.

Comment (tomczoink) [inline content/api-streamer/producer.textile]: ` give it's consumers` to `give its consumers`

File: content/api-streamer/producer.textile [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+
+<a href="/images/diagrams/api-streamer-products.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-products.png" style="width: 100%" alt="Managing data streams using products">
+</a>
+
+h3(#distribute). Distribute
+
+After you've set up products to manage the data within your apps, it's time to distribute them to the consumers of data. You can do this by setting up subscriptions in your API Streamer producer dashboard.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: `them to the consumers of data.` => `them to your consumers` or something.

File: content/api-streamer/producer.textile [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+
+Creating new subscriptions is an easy way to give access to one or more of your products to the intended consumers of your data. At the time of setting up subscriptions, you can also specify how your consumers will be charged for the data they consume. You have the options for bearing all the charges yourself, letting Ably bill the consumers directly for what they consume or even set up a custom option where you incur their charges up to a certain limit and Ably will charge them directly for any messages consumed over that limit.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: `At the time of setting up subscriptions` => `When setting up a subscription`.

File: content/api-streamer/producer.textile [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+
+<a href="/images/diagrams/api-streamer-subscriptions.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-subscriptions.png" style="width: 100%" alt="Distributing your products to consumers">
+</a>
+
+After you've set up new subscriptions, your intended consumers will automatically receive an invitation via email to use the data streams they've been given access to, with the prcing agreements you've specified in the subscription.
```
Comment (tomczoink) [inline content/api-streamer/producer.textile]: `After you've set up new subscriptions` to `After you've set up your subscription`

Comment (tomczoink) [reply content/api-streamer/producer.textile]: `prcing` to `pricing`.

File: content/api-streamer/consumer.textile [hunk 18]
```diff
@@ -0,0 +1,98 @@
+---
+title: Consumer
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: The title shows up in the sidebar. This file will be visually under 'API Streamer' :)

File: content/api-streamer/consumer.textile [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+section: api-streamer
+index: 68
+---
+
+The "Ably API Streamer":https://ably.io/api-streamer makes it easy and cost-effective for organizations and developers to offer realtime APIs, at any scale. It provides a unified solution for publishing, managing and distributing realtime data via open protocol APIs, all built on the "global Ably Data Stream Network":https://www.ably.io/dsn. API Streamer, amongst other things, is being used to share live sports scores and financial market data, mobility IoT data such as GPS locations, and SaaS integration updates in real time.
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: > open protocol APIs link to /adapters perhaps?
Linked the open-policy page, thanks.

File: content/api-streamer/consumer.textile [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+
+You can consume one or more data streams using the API Streamer either following a direct invitation from a data producer (mostly applicable for commercial and private streams) or by subscribing to an open data stream of a product listed on the "Ably Hub":https://www.ably.io/hub, a marketplace for streaming data sources. You can then manage these data streams to share it with your end clients. 
+
+h2(#consuming-managing-streams). Consuming and managing data streams
+
+Ably supports two models for subscribing to streams of data:
+
+* **Pull Subscription Protocols** are used by clients that connect to Ably and explicitly subscribe to streams on-demand. Protocols such as MQTT, SSE, or Ably's native protocol, are all Pull based protocols as clients initiate the connection to Ably and request a subscription to a stream of data from the Ably service
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: It's a subscription protocol, so `Ably's native Realtime libraries` :) Updated, thanks.

File: content/api-streamer/consumer.textile [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+
+* **Push Subscription Protocols** are outbound streams to third party systems that are initiated when data is published on a channel. Push Subscription protocols are often used where it is not practical to consume data on-demand, typically when the volumes of messages or channels are high, and complexity for the subscriber increases. Ably's Push Subscription Protocols are offered through the "Ably Reactor service":https://www.ably.io/reactor, and enable developers to stream data into their streaming or queuing services like Kafka, AMQP (RabbitMQ) or Amazon SQS/Kinesis, or to trigger serverless functions across all popular cloud providers.
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: Had a long discussion with Matt. Will explain offline.

File: content/api-streamer/consumer.textile [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+
+<a href="/images/diagrams/api-streamer-consumer.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-consumer.png" style="width: 100%" alt="API Streamer Consumer options">
+</a>
+
+h3(#subscribing-to-data-streams-using-ably). Subscribing to data streams using Ably's Realtime Client SDKs
+
+To start consuming data streams, you can log into the API Streamer's consumer dashboard. In the subscriptions tab, you'll see a list of all the products that you have access to. You can view any product for more information about the producer offering that product, as well as the documentation on how to use that product. One of the key things you need from the documentation is the channel names to which the data streams are being published.
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: updated

File: content/api-streamer/consumer.textile [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+
+To manage the data from the products you have access to easily, you can set up any number of apps in your apps dashboard. By default, all the Ably accounts come with two pre-loaded apps called 'Production' and 'Sandbox'. If you are just trying out the API Streamer for the first time, we recommend you use the 'Sandbox' app to get up and running quickly.  
+
+In order to use the Ably client SDKs, your client devices will need to authenticate with Ably first. You can simply use the API key which you can obtain from the app you wish to use, also called "Basic Auth mechanism":/core-features/authentication#basic-authentication or set up an auth server so your client devices can use the "Token Auth mechanism":/core-features/authentication#token-authentication. Tokens are short lived and hence more secure in cases where embedding API Keys on the client-side code isn't feasible. "Learn more about the various authentication mechanisms in Ably":/core-features/authentication.
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: The main reason for its inclusion is so they understand where to get their API keys from.

File: content/api-streamer/consumer.textile [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+
+The following code snippets show examples of subscribing to the streaming data channels using Ably's Client SDKs using the basic authentication for simplicity. Note: You can choose the language of your choice from the navbar on the top of this page to see an example in that language.
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/api-streamer/consumer.textile]: typo, fixed.

File: content/api-streamer/overview.textile (hunk 1 above)
Comment (Srushtika) [reply content/api-streamer/overview.textile]: Up for debate, will put down in the issue to discuss later.

File: content/api-streamer/overview.textile (hunk 2 above)
Comment (Srushtika) [reply content/api-streamer/overview.textile]: added in notes

File: content/api-streamer/producer.textile (hunk 10 above)
Comment (Srushtika) [reply content/api-streamer/producer.textile]: What if they subscribe to their own products? @tomczoink

Comment (tomczoink) [reply content/api-streamer/producer.textile]: Two-way communication to me implies that not only would they be publishing to a consumer, but the consumer would also be publishing to them. If you want to indicate it's possible they may want to subscribe as well that's fine, but I don't think two-way communication is the right way to describe that.

**Pull request merged**